plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/Library.java

```diff
@@ -200,12 +203,54 @@ public synchronized void resolveDependencies() throws CoreException {
         Collection<LibraryFile> dependencies = LibraryFactory.loadTransitiveDependencies(coordinates);
         transitiveDependencies.addAll(dependencies);
       }
-      this.libraryFiles = transitiveDependencies;
+      
+      this.libraryFiles = resolveDuplicates(transitiveDependencies);
       this.resolved = true;
     }
-  }
+  }  
   
   /**
+   * Strip out different versions of the same library, retaining only the most recent.
+   *
+   * @return a new list containing the most recent version of each dependency
+   */
+  public static List<LibraryFile> resolveDuplicates(List<LibraryFile> dependencies) {
+    List<LibraryFile> result = new ArrayList<>();
+    Map<String, MavenCoordinates> map = new HashMap<>();
+    for (LibraryFile file : dependencies) {
+      MavenCoordinates coordinates = file.getMavenCoordinates();
+      String key = coordinates.getGroupId() + ":" + coordinates.getArtifactId();
+      if (map.containsKey(key)) {
+        MavenCoordinates previousCoordinates = map.get(key);
+        if (newer(coordinates, previousCoordinates)) {
+          map.put(key, coordinates);
+          for (int i = 0; i < result.size(); i++) {
+            if (result.get(i).getMavenCoordinates().equals(previousCoordinates)) {
+              result.set(i, file); // replace
+            }
+          }
+        }
+      } else {
+        map.put(key, coordinates);
+        result.add(file);
+      }
+      
+    }
+    return result;
+  }
+
+  private static boolean newer(MavenCoordinates coordinates, MavenCoordinates previousCoordinates) {
+    try {
+      Version version1 = new Version(coordinates.getVersion());
```

Comment:
Given these are `_Maven_Coordinates`, perhaps better to use `org.apache.maven.artifact.versioning.ComparableVersion`?

(Could have `MavenCoordinates` implement `Comparable`?)

Reply:
Done.